applications/luci-app-shadowsocks-libev/htdocs/luci-static/resources/view/shadowsocks-libev/instances.js

```diff
@@ -0,0 +1,156 @@
+			   server it refers to.'));
+
+		s = m.section(form.GridSection);
+		s.addremove = true;
+		s.cfgsections = function() {
+			return this.map.data.sections(this.map.config)
+				.filter(function(s) { return cfgtypes.indexOf(s['.type']) !== -1; })
+				.map(function(s) { return s['.name']; });
+		};
+		s.sectiontitle = function(section_id) {
+			var s = uci.get(conf, section_id);
+			return (s ? s['.type'] + '.' : '') + section_id;
+		};
+		s._renderSectionAdd = s.renderSectionAdd;
+		s.renderSectionAdd = function(extra_class) {
+			var el = this._renderSectionAdd(extra_class),
+				optionEl = [E('option', { value: '_dummy' }, [_('-- instance type --')])];
+			cfgtypes.forEach(function(t) {
+				optionEl.push(E('option', { value: t }, [t.replace('_', '-')]));
+			});
+			var typeEl = E('select', { class: 'cbi-input-select' }, optionEl);
+			el.prepend(E('div', {}, typeEl));
+			return el;
+		};
+		s._handleAdd = s.handleAdd;
+		s.handleAdd = function(ev, name) {
+			var selectEl = ev.target.parentElement.firstElementChild.firstElementChild,
+				type = selectEl.value;
+			if (type === '_dummy')
+				return;
+			this.sectiontype = type;
+			var promise = this._handleAdd(ev, name);
+			this.sectiontype = undefined;
+			return promise;
+		};
+		s.addModalOptions = function(s, section_id, ev) {
+			var sdata = uci.get(conf, section_id),
+				stype = sdata ? sdata['.type'] : null;
+			if (stype) {
+				s.sectiontype = stype;
+				return Promise.all([
+					L.resolveDefault(fs.stat('/usr/bin/' + stype.replace('_', '-')), null),
+					network.getDevices()
+				]).then(L.bind(function(res) {
+					s.tab('general', _('General Settings'));
+					s.tab('advanced', _('Advanced Settings'));
+					s.taboption('general', form.Flag, 'disabled', _('Disable'));
+					if (!res[0]) {
+						ss.option_install_package(s, 'general');
+					}
+					ss.options_common(s, 'advanced');
+
+					if (stype === 'ss_server') {
+						ss.options_server(s, { tab: 'general' });
+						o = s.taboption('general', form.Value, 'bind_address',
+							_('Bind address'),
+							_('The address ss-server will initiate connection from'));
+						o.datatype = 'ipaddr';
+						o.placeholder = '0.0.0.0';
+						ss.values_ipaddr(o, res[1]);
+					} else {
+						ss.options_client(s, 'general', res[1]);
+						if (stype === 'ss_tunnel') {
+							o = s.taboption('general', form.Value, 'tunnel_address',
+								_('Tunnel address'),
+								_('The address ss-tunnel will forward traffic to'));
+							o.datatype = 'hostport';
+						}
+					}
+				}, this));
+			}
+		};
+
+		o = s.option(form.DummyValue, 'overview', _('Overview'));
+		o.modalonly = false;
+		o.editable = true;
+		o.rawhtml = true;
+		o._renderWidget = o.renderWidget;
+		o.renderWidget = function(section_id, option_index, cfgvalue) {
+			var sdata = uci.get(conf, section_id);
+			if (sdata) {
+				return this._renderWidget(section_id, option_index, ss.cfgvalue_overview(sdata));
```

Comment:
Please use `form.DummyValue.prototype.renderWidget.apply(this, [ section_id, option_index, ss.cfgvalue_overview(sdata) ]);` instead of caching `_renderWidget`.